### ISSUE TITLE
Did you find that it failed when you use padding="VALID"?

### ISSUE THREAD

**Issue by Zealoe** [Zardinality/TF-deformable-conv#9]
I  met a problem that when i set padding="VALID", the loss function did not converge. However, it converged when i set padding="SAME"...other config is same.
And i also dumped the offset. it was clipped by 0？？ （I didn't use tf.clip at all)
![image](https://user-images.githubusercontent.com/30797481/35766852-de922868-091a-11e8-99a9-cc57a315d0c1.png)

**Comment by Zardinality** [comment]
What `dilation_rate` did you use? I am guessing maybe because I use partial version of `GetWindowedOutputSize`. You can change all function call of `GetWindowedOutputSize` in `deform_conv.cc` like in [here](https://github.com/Zardinality/TF-deformable-conv/blob/master/lib/deform_conv.cc#L474:20) to [GetWindowedOutputSizeV2](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/common_shape_fns.h#L99:8) and rebuild.
Also, a dirty workaround would be slice your input to make its spatial dimension divisible by param.

**Comment by Zealoe** [comment]
the input feature map is 24*24, and the offset conv setting is
![image](https://user-images.githubusercontent.com/30797481/35774321-e1ee4a66-0963-11e8-8206-ecd0e27e47d0.png)

the deformable conv setting is
![image](https://user-images.githubusercontent.com/30797481/35774322-ee7746de-0963-11e8-8e4e-994f84df2cb6.png)

**Comment by Zealoe** [comment]
sorry to disturb, but when i change GetWindowedOutputSize in deform_conv.cc and rebuild, it failed when i run g++_complie.sh. So what should i do to make it work? Thanks!
![image](https://user-images.githubusercontent.com/30797481/35799745-afb6eac8-0aa1-11e8-8169-b2321b7c5712.png)

**Comment by Zardinality** [comment]
I am also baffled why it failed to build, `GetWindowedOutputSize` and `GetWindowedOutputSizeV2` are defined in the same file, and your arguments seem all right.
In fact, I was wondering, now that your spatial dimension is exactly divided by filter size, shouldn't `VALID` and `SAME` behave just the same?

**Comment by Zealoe** [comment]
What version of tensorflow do you use? I am using 1.1.0 and GetWindowedOutputSizeV2 is not defined in common_shape_fns.h ..     if i copy the file common_shape_fns.h including GetWindowedOutputSizeV2 to my tensorflow, the deformable conv can be built, but when i use it, it shows
![image](https://user-images.githubusercontent.com/30797481/36664122-99ddbe80-1adb-11e8-8e61-a0dd6cc617d2.png)

**Comment by Zardinality** [comment]
@Zealoe GetWindowedOutputSizeV2 is introduced in [this commit](https://github.com/tensorflow/tensorflow/commit/cf17ec96ed987386d73c645cd8b44aa32b7568b1#diff-2e1bb5f4fe3ad3cfe4e9f58f2c9d340a), so I guess it should be 1.3 or 1.4 at least. Good luck with it.
